src/Actions/MockActionResponse.php

```diff
@@ -108,35 +111,51 @@ public function assertVisit(string $path = '', array $options = []): self
             return $this->assertResponseType('visit');
         }
 
-        return $this->assertResponseContainsArray(['path' => $path, 'options' => array_values($options)], 'visit');
+        PHPUnit::assertArrayHasKey('visit', $this->response);
+
+        $visit = $this->response['visit'];
+
+        PHPUnit::logicalAnd(
+            PHPUnit::assertNotEmpty($visit),
+            PHPUnit::assertInstanceOf(Visit::class, $visit),
+            PHPUnit::assertEquals($options, $visit->options),
+            PHPUnit::assertEquals($path, $visit->path),
+        );
+
+        return $this;
     }
 
     /**
      * Asserts the handle response is of type "openInNewTab".
      *
-     * @param  string  $message
+     * @param null|string $path
+     * @param string $message
      * @return $this
      */
-    public function assertOpenInNewTab(string $message = ''): self
+    public function assertOpenInNewTab(?string $path = null, string $message = ''): self
```

Comment:
Why did this need to be changed?

Reply:
I guess on second look we do not need to change the signature, we could instead introduce a new method.

The problem I was trying to solve was there is no way to assert the new tab was the correct path.